[FEATURE] Dupliquer une campagne d'un simple clic (PIX-8635)

Alexandre-Monney opened the pull request:
## :unicorn: Problème
Nos prescripteurs ont énormément de campagnes à créer parfois seulement avec un PC ou un titre de campagne à changer. Au moment de la rentrée par exemple avec les professeurs SCO de chaque classe créant des campagnes 6ième, 5ième, 4ième, ou encore les enseignants du SUP avec toutes les différentes licences etc.

Suite de cette PR : 
- https://github.com/1024pix/pix/pull/6586


## :robot: Proposition

- Affichage du bouton dupliquer dans l’onglet paramètres
- mettre les boutons au bon endroit dans cet onglet
- à gauche modifier et dupliquer
- à droite le bouton archiver

## :rainbow: Remarques

La maquette prévoit l'ajout d'icônes dans les boutons "Modifier", "Dupliquer" et "Archiver" mais le `PixButtonLink` dans sa forme actuelle ne permet l'ajout d'icône.
Le style des boutons a donc été conservé.

## :100: Pour tester

- Se connecter à Pix Orga
- Aller dans les paramètres d'une campagne
- Appuyer sur le bouton Dupliquer
- Vérifier que les infos de la campagne précédente ont bien été reprises dans la copie
- 🐱

HEYGUL committed:
feat (orga): add button to duplicate displayed campaign

Co-authored-by: <alexandre.monney@pix.fr>
Co-authored-by: <vincent.hardouin@pix.fr>

HEYGUL committed:
feat (orga): handle Cancel when duplicating a campaign

when duplicating a campaign, the source parameter is filled in
in this case, when canceling, we redirect user to settings page
of the source campaign

Co-authored-by: <alexandre.monney@pix.fr>
Co-authored-by: <vincent.hardouin@pix.fr>

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr6621.review.pix.fr)
  * [App (.org)](https://app-pr6621.review.pix.org)
  * [Orga (.fr)](https://orga-pr6621.review.pix.fr)
  * [Orga (.org)](https://orga-pr6621.review.pix.org)
  * [Certif (.fr)](https://certif-pr6621.review.pix.fr)
  * [Certif (.org)](https://certif-pr6621.review.pix.org)
  * [Admin](https://admin-pr6621.review.pix.fr)
  * [API](https://api-pr6621.review.pix.fr/api/)
  * [Pix1D](https://1d-pr6621.review.pix.fr)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr6621/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr6621/environment)

HEYGUL committed:
feat (orga): adapt e2e test to use duplicate campaign button

Alexandre-Monney commented:
Quand on clic sur dupliquer une campagne et qu'on valide la création, si ensuite on veut retourner sur la liste des campagnes pour en créer une de zéro, on se retrouve avec les informations de la campagne dupliquée toujours préremplie. Est ce que c'est un comportement voulu ?

VincentHardouin commented:
> Quand on clic sur dupliquer une campagne et qu'on valide la création, si ensuite on veut retourner sur la liste des campagnes pour en créer une de zéro, on se retrouve avec les informations de la campagne dupliquée toujours préremplie. Est ce que c'est un comportement voulu ?

Ah oui la source est conservée, ça ne devrait pas :/

HEYGUL committed:
fix (orga): reset query parameter to avoid duplicating campaign

when a query parameter has been set, it persists accross navigation to
the same route

when exiting the route, we reset the query parameter so creating a
campaign from Create button starts with a blank form

Co-authored-by: <alexandre.monney@pix.fr>
Co-authored-by: <miranda.lin-guignard@pix.fr>
Co-authored-by: <vincent.hardouin@pix.fr>
Co-authored-by: <xavier.carron@pix.fr>

HEYGUL committed:
chore (orga): remove extra console.log
